### PR TITLE
Group project : Catalog-of-my-things

### PR DESCRIPTION
**Pull request by hakichris**
[Video Walkthrough](https://drive.google.com/file/d/1dGZ8XjP7qJ_J2JurZg3jBi_AEB2kRVEU/view?usp=share_link)
This project is a group project, a team of three people.
The project is implemented using Ruby. 
During this week, here is the list of things we have done.

- The project is based on the given UML diagram
- List all books
- List all music albums
- List of games
- List all genres (e.g 'Comedy', 'Thriller')
- List all labels (e.g. 'Gift', 'New')
- List all authors (e.g. 'Stephen King')
- Add a book
- Add a music album
- Add a game
- Create unit tests for all implemented methods
- Write/Read from files
- Create a console app entry-point 
- Create a database analogical to the structure of the application
- Follow General Requirements
- Create a professional Readme file
- Provide a demo video (cf. at the top)

That's it 🤷🏽 
Happy Review! 👁️‍🗨️

### CODE REVIEW

File: spec/music_test_spec.rb [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    it 'Should list all music albums' do
+      @app.music_albums = [@music_album_one]
+      expect do
+        @app.list_music_album
+      end.to output {
+        "_____LIST OF MUSIC ALBUM_____\n\n0- Music id: #{@music_album_one.id} - is published on 2028-10-07\n"
+      }.to_stdout
+    end
+  end
```
Comment (danmainah) [inline spec/music_test_spec.rb:48]: - [ ] @team great work on your tests they are all passing, however, I didn't see tests for the `to_json(*args)` and `self.from_json(string)` methods in your music class, I would recommend you add them.
- [ ] @team I didn't manage to see tests for the Item, App, Source, Bookdata, and GameActions classes. it was part of the project requirement to implement unit tests for all methods in your classes as shown in the highlighted section below
image

File: database/schema.sql [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  -- inherited from item table
+  id int GENERATED BY DEFAULT AS IDENTITY,
+  publish_date date NOT NULL,
+  archived boolean
+) INHERIT(item);
+
+-- Create table genre
+CREATE TABLE genre(
+  id int GENERATED BY DEFAULT AS IDENTITY,
+  name varchar(255)
```
Comment (danmainah) [inline database/schema.sql:29]: - [ ] @team great work on your schema, however, you forgot to add a foreign key referencing the item table like you did in the authors table.

File: database/schema.sql [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  publish_date DATE,
+  publisher VARCHAR(50),
+  cover_state VARCHAR(50),
+  Item_id INTEGER REFERENCES item(id),
+) 
+
+CREATE TABLE labels(
+  id SERIAL  primary key 
+  title varchar(50),
+  color varchar(50)
```
Comment (danmainah) [inline database/schema.sql:61]: @team great work on your schema, however, you forgot to add a foreign key referencing the item table like you did in the authors table.

**Review by danmainah — CHANGES_REQUESTED** [review]
Hi @hakichris @Marcraphael12 and @rica213 , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️  Good design on the user interface
✔️  Good professional PR and Description
✔️ Professional README file.


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: database/schema.sql (hunk 2 above)
Comment (hakichris) [reply database/schema.sql:29]: Thank you @danmainah, we rectified according to your recommendation

File: database/schema.sql (hunk 3 above)
Comment (hakichris) [reply database/schema.sql]: Thank you @danmainah, we rectified according to your recommendation

File: spec/music_test_spec.rb (hunk 1 above)
Comment (hakichris) [reply spec/music_test_spec.rb:51]: Thank you @danmainah, we rectified according to your recommendation

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @Marcraphael12, @hakichris, and @rica213👋, 

### You did a marvelous job on the Ruby group capstone project and this is one of the best projects I have seen so far!
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## To Highlight 🎉
- All tests are passed. ✔️
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Highly organized code. ✔️
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**